packages/story-editor/src/components/form/tags/input.js

```diff
@@ -135,48 +171,110 @@ function Input({
           setIsInputFocused(false);
         },
       }),
-      [onUndo]
+      [onUndo, suggestedTerms]
     );
 
   const renderedTags = tagBuffer || tags;
+
+  const handleTagSelectedFromSuggestions = useCallback((e, selectedValue) => {
+    e.preventDefault();
+    dispatch({ type: ACTIONS.SUBMIT_VALUE, payload: selectedValue });
+    inputRef?.current.focus();
+  }, []);
```

Comment:
Might be nice to store some state for if the suggestions list is open so it closes immediately on selection. something like:
```js
  const [isSuggestionsOpen, setIsSuggestionsOpen] = useState(false);
...
  const handleTagSelectedFromSuggestions = useCallback((e, selectedValue) => {
    e.preventDefault();
    dispatch({ type: ACTIONS.SUBMIT_VALUE, payload: selectedValue });
    inputRef?.current.focus();
    setIsSuggestionsOpen(false)
  }, []);
...
  const suggestionLength = suggestedTerms.length;
  useEffect(() => {
    setIsSuggestionsOpen(suggestionLength > 0);
  }, [suggestionLength]);
```

Reply:
I like this idea - there's some time between selection and closing due to the debounce